server/plugins/task-plugin-ob-mysql/src/main/java/com/oceanbase/odc/plugin/task/obmysql/partitionplan/datatype/BasePartitionKeyDataTypeFactory.java

```diff
@@ -70,8 +70,8 @@ public DataType generate() {
          * assume that the partition key is a column
          */
         Map<String, DBTableColumn> colName2Col = this.dbTable.getColumns().stream()
-                .collect(Collectors.toMap(c -> unquoteIdentifier(c.getName()), c -> c));
-        DBTableColumn column = colName2Col.get(unquotedPartitionKey);
+                .collect(Collectors.toMap(c -> unquoteIdentifier(c.getName()).toLowerCase(), c -> c));
```

Comment:
we may modify `OBMySQLAutoPartitionExtensionPoint#unquoteIdentifier` directly to fix this~

Reply:
Collectors.toMap may change to use TreeMap with CASE_INSENSITIVE Comparator